src/components/MovieForm/MovieForm.tsx

```diff
@@ -1,53 +1,58 @@
 import styles from "../MovieForm/MovieForm.module.css";
-import React, {FormEvent} from "react";
+import React from "react";
 import {MovieDetailsData} from "../../types/MovieDetailsData";
-import {GenreType} from "../../constants/GenreType";
 import InputField from "../InputField/InputField";
 import GenreSelect from "../GenreSelect/GenreSelect";
+import {useForm, SubmitHandler} from "react-hook-form";
+import {MovieDetailsPost} from "../../types/MovieDetailsPost";
+import {z} from "zod";
+import {movieFormSchema} from "../../types/MovieFormSchema";
+import {GenreType} from "../../constants/GenreType";
+import {zodResolver} from "@hookform/resolvers/zod";
 
 interface MovieFormProps {
     initialMovieInfo?: MovieDetailsData | null,
-    handleSubmit: (movieData: MovieDetailsData) => void;
+    onMovieSubmit: (movieData: MovieDetailsPost) => void;
 }
 
-const MovieForm: React.FC<MovieFormProps> = ({ initialMovieInfo, handleSubmit }) => {
-    function onSubmit(e: FormEvent<HTMLFormElement>) {
-        e.preventDefault();
-        const form = e.target as HTMLFormElement;
-        const formData = new FormData(form);
-        const formDataObject = Object.fromEntries(formData) as Record<string, string>;
-
-        const genres = formData.getAll("genres") as string[] as GenreType[];
-
-        const id = initialMovieInfo?.id ?? Math.floor(Math.random() * 10000);
+type MovieFormSchema = z.infer<typeof movieFormSchema>
 
+const MovieForm: React.FC<MovieFormProps> = ({initialMovieInfo, onMovieSubmit}) => {
+    const {
+        register,
+        handleSubmit,
+        formState: {errors, isSubmitting},
+    } = useForm<MovieFormSchema>({
+        resolver: zodResolver(movieFormSchema)
+    });
+    const onSubmit: SubmitHandler<MovieFormSchema> = async (data) => {
         const movieData = {
-            id,
-            poster_path: initialMovieInfo?.poster_path || "",
-            title: formDataObject.title,
-            release_date: formDataObject.release_date,
-            movieUrl: formDataObject.movieUrl,
-            vote_average: formDataObject.vote_average,
-            genres,
-            runtime: formDataObject.runtime,
-            overview: formDataObject.overview,
+            id: initialMovieInfo?.id,
+            poster_path: data.poster_path,
+            title: data.title,
+            release_date: data.release_date,
+            vote_average: data.vote_average,
+            genres: data.genres as GenreType[],
+            runtime: data.runtime,
+            overview: data.overview,
         };
 
-        handleSubmit(movieData);
+        await onMovieSubmit(movieData);
     }
```

Comment:
Can we do here destructuring? 
![image](https://github.com/user-attachments/assets/17cddbb9-a422-4b29-ac44-287d43ff4da3)

and after that do like this: 
`
const movieData = {
id, 
title, 
...
runtime, 
}
`